pkg/envoy/eds/response.go

```diff
@@ -35,7 +35,8 @@ func fulfillEDSRequest(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, re
 		return nil, fmt.Errorf("Endpoint discovery request for proxy %s cannot be nil", proxy.Identity)
```

Comment:
Let's remove fulfillEDSRequest and generateEDSConfig, and merge them directly into NewResponse

Reply:
similarly i think we can also collapse `getUpstreamEndpointsForProxyIdentity` into NewResponse